evm/deploy-amplifier-gateway.js

```diff
@@ -167,11 +168,22 @@ async function deploy(config, chain, options) {
         return;
     }
 
+    const {
+        axelar: { contracts, network, axelarId },
+    } = config;
+    const {
+        Router: { address: routerAddress },
+    } = contracts;
     contractConfig.deployer = wallet.address;
     const domainSeparator = await getDomainSeparator(config, chain, options);
+    const expectedDomainSeparator = calculateDomainSeparator(axelarId, routerAddress, network);
     const minimumRotationDelay = Number(options.minimumRotationDelay);
     const salt = options.salt || '';
 
+    if (domainSeparator !== expectedDomainSeparator) {
+        throw new Error(`unexpected domain separator (want ${expectedDomainSeparator}, got ${domainSeparator})`);
+    }
```

Comment:
check this in getDomainSeparator perhaps. The check should be made if domainSeparator isn't being overridden, otherwise the override isn't useful if a deterministic value is used